packages/providers/upload-cloudinary/src/index.ts

```diff
@@ -62,32 +61,41 @@ module.exports = {
 
         if (file.stream) {
           file.stream.pipe(uploadStream);
-        } else {
+        } else if (file.buffer) {
           intoStream(file.buffer).pipe(uploadStream);
+        } else {
+          throw new Error('Missing file stream or buffer');
         }
       });
+    };
 
     return {
-      uploadStream(file, customConfig = {}) {
+      uploadStream(file: File, customConfig = {}) {
         return upload(file, customConfig);
       },
-      upload(file, customConfig = {}) {
+      upload(file: File, customConfig = {}) {
         return upload(file, customConfig);
       },
-      async delete(file, customConfig = {}) {
+      async delete(file: File, customConfig = {}) {
         try {
-          const { resource_type: resourceType, public_id: publicId } = file.provider_metadata;
-          const response = await cloudinary.uploader.destroy(publicId, {
+          const { resource_type: resourceType, public_id: publicId } = file.provider_metadata ?? {};
+          const deleteConfig = {
+            resource_type: (resourceType || 'image') as string,
             invalidate: true,
-            resource_type: resourceType || 'image',
             ...customConfig,
-          });
+          };
+
+          const response = await cloudinary.uploader.destroy(`${publicId}`, deleteConfig);
 
           if (response.result !== 'ok' && response.result !== 'not found') {
             throw new Error(`Error deleting on cloudinary: ${response.result}`);
```

Comment:
maybe we should change this error too
```suggestion
            throw new Error(`${response.result}`);
```
so it does not colide with the catch error message below. wdyt?

Reply:
Sure we can do that